### PR TITLE
Companion code for Scala-79

### PR DESCRIPTION
**Pull request by oforero**
Ready for review.

The warnings are on purpose since I would like the code to be really explicit and not take any shortcuts that makes it hard of readers to understand.

### CODE REVIEW

File: scala-core-oop/src/main/scala/com/baeldung/scala/bounds/viewbound.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    def less(other: T): Boolean
+  }
+
+  implicit val intToOrder: Int => Order[Int] = x => new Order[Int](x) {
+    override def less(other: Int): Boolean = me < other
+  }
+
+  def maximum[A <% Order[A]](a: A, b: A): A = {
+    val toOrder = implicitly[A => Order[A]]
+    if(toOrder(a).less(b)) b else a
```
Comment (rpvilao) [inline scala-core-oop/src/main/scala/com/baeldung/scala/bounds/viewbound.scala:14]: space in if sentence

File: scala-core-oop/src/main/scala/com/baeldung/scala/bounds/richobject.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  implicit val intOrder: Order[Int] = new Order[Int] {
+    override def less(me: Int, other: Int): Boolean = me < other
+  }
+
+  def maximum[A:Order](a: A, b: A): A = {
+    val ord = implicitly[Order[A]]
+    import ord._
+    if(a < b) b else a
```
Comment (rpvilao) [inline scala-core-oop/src/main/scala/com/baeldung/scala/bounds/richobject.scala:19]: space in if sentence

File: scala-core-oop/src/main/scala/com/baeldung/scala/bounds/implicitobject.scala [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  abstract class Order[T] {
+    def less(me: T, other: T): Boolean
+  }
+
+  implicit val intOrder: Order[Int] = new Order[Int] {
+    override def less(me: Int, other: Int): Boolean = me < other
+  }
+
+  def maximum[A](a: A, b: A)(implicit ord: Order[A]): A = {
+    if(ord.less(a, b)) b else a
```
Comment (rpvilao) [inline scala-core-oop/src/main/scala/com/baeldung/scala/bounds/implicitobject.scala:13]: space in if sentence

File: scala-core-oop/src/main/scala/com/baeldung/scala/bounds/implicitconversion.scala [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  abstract class Order[T](val me: T) {
+    def less(other: T): Boolean
+  }
+
+  implicit val intToOrder: Int => Order[Int] = x => new Order[Int](x) {
+    override def less(other: Int): Boolean = me < other
+  }
+
+  def maximum[A](a: A, b: A)(implicit toOrder: A => Order[A]): A = {
+    if(toOrder(a).less(b)) b else a
```
Comment (rpvilao) [inline scala-core-oop/src/main/scala/com/baeldung/scala/bounds/implicitconversion.scala:13]: space in if sentence

File: scala-core-oop/src/main/scala/com/baeldung/scala/bounds/conversion.scala [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  abstract class Order[T](val me: T) {
+    def less(other: T): Boolean
+  }
+
+  val intToOrder: Int => Order[Int] = x => new Order[Int](x) {
+    override def less(other: Int): Boolean = me < other
+  }
+
+  def maximum[A](a: A, b: A)(toOrder: A => Order[A]): A = {
+    if(toOrder(a).less(b)) b else a
```
Comment (rpvilao) [inline scala-core-oop/src/main/scala/com/baeldung/scala/bounds/conversion.scala:13]: space in if sentence

File: scala-core-oop/src/main/scala/com/baeldung/scala/bounds/contextbound.scala [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    def less(me: T, other: T): Boolean
+  }
+
+  implicit val intOrder: Order[Int] = new Order[Int] {
+    override def less(me: Int, other: Int): Boolean = me < other
+  }
+
+  def maximum[A:Order](a: A, b: A): A = {
+    val ord = implicitly[Order[A]]
+    if(ord.less(a, b)) b else a
```
Comment (rpvilao) [inline scala-core-oop/src/main/scala/com/baeldung/scala/bounds/contextbound.scala:14]: space in if setence

**Comment by oforero** [comment]
@rpvilao I Have corrected my local setup and now run the automatic formatter when saving. 
I also notice there is a test in code that it is failing from time to time, I think my small change will ensure the test is more reliable.

**Pull request merged**